torch/distributed/distributed_c10d.py

```diff
@@ -4780,6 +4780,10 @@ def barrier(
         None, if not async_op or if not part of the group
 
     .. note:: `ProcessGroupNCCL` now blocks the cpu thread till the completion of the barrier collective.
+    .. warning:: `ProcessGroupNCCL` implements barrier as an all_gather of a 1-element tensor. This tensor will be
+       allocated on the device specified by 'device_ids' arg if specified, or the device set with
```

Comment:
`device_ids` sounds like a plural, i.e. could be list/tuple. Do you want to clarify that it will be allocated on first (or one of )devices listed in  `device_ids`

> `the device set with `torch.cuda.set_device`
This sounds a bit confusing to me, especially for multithreaded apps, where each thread can have their own default device. May be it should say something or `or on the current device, which for given thread can be queried using `torch.cuda.get_device` or altered using `torch.cuda.set_device` API or `torch.device` context manager

Reply:
turns out i was incorrect about my understanding of how barrier selects device, it has been improved since i last saw it. I updated the doc to reflect what I see in `ProcessGroupNCCL::guessDeviceId`

Reply:
I think this is vestige of when we used to support multigpu collectives (one process using multiple GPUs) e.g. https://github.com/pytorch/pytorch/pull/85961. But now, our main assumption is 1 process = 1 GPU